Sources/StreamVideo/WebRTC/Statistics/Statistics+Convenience.swift

```diff
@@ -89,13 +89,16 @@ extension RTCStatistics: StreamStatisticsProtocol {}
 
 extension RTCStatisticsReport {
     func jsonString(for type: PeerConnectionType) -> String? {
-        let statsKey = type == .publisher ? "publisherStats" : "subscriberStats"
         var statsArray = [Any]()
-        for (key, value) in statistics {
-            statsArray.append([key: value.values])
+        for (_, value) in statistics {
+            var entry: [String: Any] = ["type": value.type, "timestamp": value.timestamp_us, "id": value.id]
+            for (key, value) in value.values {
+                entry[key] = value
+            }
+            statsArray.append(entry)
         }
         if let json = try? JSONSerialization.data(
```

Comment:
I haven't followed the stats implementation. Why are we using JSONSerialization over Codable?

Reply:
Is it worth adding tests for this code?

Reply:
- We can't use Codable - this is the WebRTC stats object that we convert to string for the stats report. 
- `init` of a stats report is not possible, so we also can't write tests. If you have any ideas how to do that, let me know.

Reply:
The types are still Codable just in the archaic format. We can live with it for sure but it's worth discussing because JSONSerializsation is much less performant that JsonEnconder.

`RTCStatisticsReport` and `RTCStatistic` cannot be directly constructed but they can be subclassed and mocked.

Reply:
their properties are readonly as well. I ended up using your mock approach to test it.

Reply:
I love it! Thank you :)